packages/reaction-core/client/templates/products/productDetail/productDetail.js

```diff
@@ -161,37 +161,41 @@ Template.productDetail.events({
         if (cartId && productId) {
           count = ReactionCore.Collections.Cart.findOne(cartId).cartCount() || 0;
 
-          Meteor.call("cart/addToCart", cartId, productId, currentVariant, quantity, function (error) {
-            let address;
-            if (!error && count === 0) {
-              address = Session.get("address");
-              if (!address) {
-                return locateUser();
+          Meteor.call("cart/addToCart", cartId, productId, currentVariant,
+            quantity, function (error) {
+              let address;
+              if (!error && count === 0) {
+                // todo understand that's going on here
+                //address = Session.get("address");
```

Comment:
I think that originally the behaviour was that if you selected an address during an order, that did not save it as your default address in your addressBook - or in the cart.  It was just the selected address for that checkout session.  I think this is probably not needed with the correct behaviour being that the cart specific addressbook selection is now stored in the cart, and the user's default addressbook is in the accounts.